addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1107 +1,1031 @@
-        this.title = params.title || '';
-        this.fields = params.fields || {};
-        this.disableLinking = params.disableLinking;
-
-        this.chart = null;
-        this.chartId = _.uniqueId('chart');
-        this.$legendTooltip = null;
-        this.$tooltip = null;
-    },
-    /**
-     * Chart.js does not need the canvas to be in dom in order
-     * to be able to work well. We could avoid the calls to on_attach_callback
-     * and on_detach_callback.
-     *
-     * @override
-     */
-    on_attach_callback: function () {
-        this._super.apply(this, arguments);
-        this.isInDOM = true;
-        this._render();
-    },
-    /**
-     * @override
-     */
-    on_detach_callback: function () {
-        this._super.apply(this, arguments);
-        this.isInDOM = false;
-    },
-
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
-
-    /**
-     * This function aims to remove a suitable number of lines from the tooltip in order to make it reasonably visible.
-     * A message indicating the number of lines is added if necessary.
-     *
-     * @private
-     * @param {Number} maxTooltipHeight this the max height in pixels of the tooltip
-     */
-    _adjustTooltipHeight: function (maxTooltipHeight) {
-        var sizeOneLine = this.$tooltip.find('tbody tr')[0].clientHeight;
-        var tbodySize = this.$tooltip.find('tbody')[0].clientHeight;
-        var toKeep = Math.floor((maxTooltipHeight - (this.$tooltip[0].clientHeight - tbodySize)) / sizeOneLine) - 1;
-        var $lines = this.$tooltip.find('tbody tr');
-        var toRemove = $lines.length - toKeep;
-        if (toRemove > 0) {
-            $lines.slice(toKeep).remove();
-            var tr = document.createElement('tr');
-            var td = document.createElement('td');
-            tr.classList.add('o_show_more');
-            td.innerHTML = _t("...");
-            tr.appendChild(td);
-            this.$tooltip.find('tbody').append(tr);
+    'use strict';
+
+    const config = require('web.config');
+    const core = require('web.core');
+    const dataComparisonUtils = require('web.dataComparisonUtils');
+    const fieldUtils = require('web.field_utils');
+    const OwlAbstractRenderer = require('web.AbstractRendererOwl');
+    const { sortBy } = require('web.utils');
+
+    const _t = core._t;
+    const DateClasses = dataComparisonUtils.DateClasses;
+    const qweb = core.qweb;
+    const { useRef } = owl.hooks;
+
+    const NO_DATA = [_t('No data')];
+    NO_DATA.isNoData = true;
+
+    const COLORS = ["#1f77b4", "#ff7f0e", "#aec7e8", "#ffbb78", "#2ca02c", "#98df8a", "#d62728",
+        "#ff9896", "#9467bd", "#c5b0d5", "#8c564b", "#c49c94", "#e377c2", "#f7b6d2",
+        "#7f7f7f", "#c7c7c7", "#bcbd22", "#dbdb8d", "#17becf", "#9edae5"];
+    const COLOR_NB = COLORS.length;
```

Comment:
Not a fan of the variable name, so I wouldn't define one (it is not really necessary here)